src/openepd/model/common.py

```diff
@@ -117,6 +117,18 @@ class Ingredient(BaseOpenEpdSchema):
     )
     citation: str | None = pyd.Field(default=None, description="Text citation describing the data source ")
 
+    @pyd.root_validator(skip_on_failure=True)
+    def _validate_evidence(cls, values: dict[str, Any]) -> dict[str, Any]:
+        # gwp_fraction should be backed by some type of evidence (fraction coming from product EPD etc) to be accounted
+        # for in the calculation of uncertainty
+        if values.get("gwp_fraction"):
```

Comment:
Missing "gwp_fraction" key and `None` value means different things when `PATCH` request. Are you taking this into account?

Reply:
We pass Ingredients as an array, and there is no identifier there, so we are only able to replace the entire ingredients array and not patch them.